src/pages/dashboard/Explore.tsx

```diff
@@ -49,46 +67,70 @@ const StyledList = styled(List)`
 
 const Explore = () => {
   const firstPage = 1;
+  const [page, setPage] = useState(firstPage);
   const location = useLocation();
-  const [query, setQuery] = useState<null | QueryObject>(null);
+
+  const [queryObject, setQueryObject] = useState(emptyQueryObject);
   const [isSearching, setIsSearching] = useState(false);
   const [searchResult, setSearchResult] = useState<SearchResult>();
   const [resources, setResources] = useState<Resource[]>([]);
   const { t } = useTranslation();
   const [searchError, setSearchError] = useState(false);
-  const [page, setPage] = useState(firstPage);
   const history = useHistory();
 
   const handlePaginationChange = (event: React.ChangeEvent<unknown>, value: number) => {
     setPage(value);
-    const searchParams = new URLSearchParams(location.search);
-    value === firstPage ? searchParams.delete('page') : searchParams.set('page', '' + value);
-    history.replace(`?${searchParams.toString()}`);
+    setQueryObject((prevState) => ({
+      ...prevState,
+      offset: (Number(value) - 1) * NumberOfResultsPrPage,
+      queryFromURL: false,
+    }));
   };
 
   useEffect(() => {
-    const searchTerm = new URLSearchParams(location.search);
-    const pageTerm = searchTerm.get('page');
-    let offset = 0;
-    if (pageTerm) {
-      setPage(+pageTerm);
-      offset = (+pageTerm - 1) * NumberOfResultsPrPage;
-    } else {
-      setPage(firstPage);
+    const createQueryFromUrl = () => {
+      const searchTerms = new URLSearchParams(location.search);
```

Comment:
litt for stor PR til at jeg klarer å henge med i alt som skjer, men her har vi i hvert fall ulik tilnærming.
Vi henter searchTerms på samme måte som her, men gjør det på "toppnivå" i render, så har vi en useEffect som kjører nytt søk hver gang searchTerms endrer seg.

```
const MyComp = () => {
   const searchTerms = new URLSearchParams(location.search);
   useEffect(()=> {
      // Perform new search
   , [searchTerms]}
   
   return (...)
```

Reply:
når vi vil søke på nytt så endrer vi bare params i URL, så fanges det opp i useeffect og vi søker på nytt. det forutsetter at du bare vil søke når brukerene aktivt velger å trykke "søk" eller noe da, for det blir litt rart om URLen endres samtidig som man skriver (syns jeg)

Reply:
Jeg fikk problemer når jeg brukte urlen som state for søkekomponentene - ettersom jeg ikke ville ha alle parametre i urlen. offset/limit/. Parsing av facets ble også mer komplekst.

Reply:
ja, vi har bare en søkestreng nå, så det gjør ting en del enklere...